File: Strategy02/Strategy02/ViewController.m
//
//  ViewController.m
//  Strategy02
//
//  Created by FishYu on 2019/8/1.
//  Copyright © 2019 FY_Developer. All rights reserved.
//

#import "ViewController.h"
#import "Validator.h"

@interface ViewController () <UITextFieldDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.AlaphaTextField.inputValidator = [[AlphaInputValidator alloc] init];
    self.numericTextField.inputValidator = [[NumericValidator alloc] init];
    // Do any additional setup after loading the view.
}

- (void)textFieldDidEndEditing:(UITextField *)textField {
    if ([textField isKindOfClass:[CustomTextField class]]) {
        [(CustomTextField *)textField vaild];
    }
}


@end

File: Observe02/Observe02/ViewController.swift
//
//  ViewController.swift
//  Observe02
//
//  Created by FishYu on 2019/8/2.
//  Copyright © 2019 FY_Developer. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()

        let observed = MyObjectToObserve()
        let observer = MyObserver(object: observed)
        observed.updateDate() // 出发观察者
        
    }

}

